### PR TITLE
Elide logits if the logits pointer parameter is NULL

### PR DESCRIPTION
**Pull request by LoganDark**
There is no need to calculate logits when it is not necessary, so perform some trickery to avoid calculating that part of the cgraph if logits is not actually going to be used.

Depends on #103

Discussed on https://github.com/saharNooby/rwkv.cpp/issues/106#issuecomment-1605890591

### CODE REVIEW

File: rwkv.cpp [hunk 1]
```diff
@@ -1568,6 +1601,16 @@ bool rwkv_eval(struct rwkv_context * ctx, const uint32_t token, const float * st
 
     rwkv_set_inputs(ctx, state_in);
     ggml_set_i32(ctx->serial_graph.tokens, token);
+
+    // Short circuit computation of logits if nobody actually cares
```
Comment (saharNooby) [inline rwkv.cpp:1605]: I suggest
`// Short circuit computation of logits if they are not needed`

**Comment by saharNooby** [comment]
Please add info about possibility of skipping logits calculation into documentation in `rwkv.h`. The [commit description](https://github.com/saharNooby/rwkv.cpp/pull/107/commits/185243a344fbb7c891e12eb17b345123e78a16a6) is already well written, and can be copy-pasted into the doc.

I would also suggest implementing it in PyTorch wrapper; but this is optional, especially given that the wrapper will be rewritten soon.

File: rwkv.cpp (hunk 1 above)
Comment (LoganDark) [reply rwkv.cpp:1605]: But my comment is funnier~

**Review by saharNooby — APPROVED** [review]
